### ISSUE TITLE
<Link> and <Button to="..."> Component always opens in new tab, regardless of `external` prop on some stores

### ISSUE THREAD

**Issue by alex-a-pereira** [Shopify/ui-extensions#1167]
## Please list the package(s) involved in the issue, and include the version you are using

```
    "@shopify/app": "^3.43.0",
    "@shopify/checkout-ui-extensions-react": "^0.24.0",
    "@shopify/cli": "^3.43.0",
    "react": "^17.0.0"
```

## Describe the bug

When rendering a `<Link to="..." external={false}>` component, we expect that the link **always** opens in a new tab. 

This only happens on our production store - the development/test store, which runs the same exact code, works properly.

This also happens for the `<Button>` component, which doesn't have an `external` prop. If the `to="..."` prop is set on a Button, then once pressed:
- in the development/test store, it opens in the same tab
- in the production store, it opens in a new tab.

## Steps to reproduce the behavior:

1. Create a checkout UI extension (typescript react) with the sample code shown below
2. Install app to store
3. You may or may not observe that the link opens in a new tab, even though the `external` prop is set to `false`

Sample Code
This is the code snippet that shows the issue. When the link is pressed, it **should** navigate the user (within the same tab) to the same URL with a query param `?locale={selected_language}` so that the checkout copy is displayed in the correct language. This code works properly on one store, but opens in a new tab on our production (customer facing) store.

```
const linkEnglish = `https://${myshopifyDomain}/checkout?locale=en`
  const linkSpanish = `https://${myshopifyDomain}/checkout?locale=es`
  const linkContent = languageCode === ENGLISH_CODE ? 'Cambiar idioma a Español' : 'Change language to English'

  return (
    <View>
      <Heading>{translate('languagePreference')}: {languageCode === ENGLISH_CODE ? 'English' : 'Español'}</Heading>
      <BlockSpacer spacing='base' />
      <InlineLayout columns='auto' spacing='loose'>
        {
          languageCode === ENGLISH_CODE ? (
            <Link onPress={handleOnPress} to={linkSpanish} external={false}>{linkContent}</Link>
          ) : (
              <Link onPress={handleOnPress} to={linkEnglish} external={false}>{linkContent}</Link>
            )
        }
      </InlineLayout>
    </ View>
```

## Expected behavior

When the Link's `external` prop is set to `false`, the link should always open within the same tab on every store.

## Screenshots

On our staging store, no `target="_blank"` is added to the a tag.

![image](https://github.com/Shopify/ui-extensions/assets/36137079/7cadde84-8952-4a2f-8dbc-f94eb262d24f)

But on our production store, `target="_blank"` is added to the a tag for some reason.

![image](https://github.com/Shopify/ui-extensions/assets/36137079/80a00b77-d76b-4050-8cab-28a9f2e3514b)

**Comment by jun-shop** [comment]
Thanks for reporting - was able to repro on shopify/ui-extensions-react v.2023.7.0 in dev mode on my test store. I'll triage this and have someone look into it

**Comment by StepanMynarik** [comment]
Same problem here.

I need to navigate to a separate page as a workaround for iframe throwing "Unsupported component" error in the console.
We need our customers to be able to use a map widget when selecting a pickup point.

Please fix this

**Comment by LucasLacerdaUX** [comment]
@StepanMynarik Thanks for reporting it. Is the issue you're dealing with the same as described in the original post above, as in not being able to open content from `Button` and `Link` in same tab? I see you mentioned `iframe` and using a map widget, so just wanted some more details to confirm if it's really the same cause.

Also, could you please share whether you are running the latest versions of `shopify/ui-extensions` / `shopify/ui-extensions-react`? 

I was initially unable to reproduce the main issue described above, but I can have another look into this problem if it is still happening.

Thank you.

**Comment by StepanMynarik** [comment]
@LucasLacerdaUX  
Yep, this is about both **Link** and **Button** components always generating with 'target="_blank"' no matter what value you specify for the component's 'external' property.

Yep, using version '2023.10.x' for both packages.

This is freshly created project. Everything updated.

**Comment by StepanMynarik** [comment]
@LucasLacerdaUX 
I see your point there.

We use the _Link_ component currently to take the user to a separate page on our store that has custom liquid template with widget for shipping service pickup point selection map.

This is a workaround for checkout extensions currently not supporting _iframes_.

The _Button_ component looks better for this task. User almost doesn't register this workaround as the widget map would usually be opened in almost fullscreen modal on the same page. When he selects a pickup point, he is navigated back to the checkout page immediately.

Link component works, but as I said, button is more noticable and fitting for this task.

Yes, it is a workaround, but a necessary workaround.
If support for iframes is added in the future, the button will stop navigating to a new page and it will just open a Modal component with iframe in it instead.

Perceivably almost no difference for most users. That is the plan at least :)

**Comment by sabir-nagarro** [comment]
Sa

> This needs to be looked at on a deeper level. For example, our checkout and cart don't live on the same domains. So, on our Checkout, we want to have a `Back to Cart` button but because it lives on a different domain, your "links" are incurring a security penalty. There needs to be a workaround for cases like this.

same issue we are facing.
